### DISCUSSION TITLE
Question for the base url of OpenAI

### DISCUSSION THREAD

**Discussion by intotherealworld** [spring-projects/spring-ai#463] [Q&A] [answered]
Hi, I have a question on base url.

The base url for OpenAI contains the version in the python package [like this](https://github.com/openai/openai-python/blob/e8e5a0dc7ccf2db19d7f81991ee0987f9c3ae375/src/openai/_client.py#L110):
> https://api.openai.com/v1

But, Spring AI has version in uri path as a fixed value [like this](https://github.com/spring-projects/spring-ai/blob/c5f35029f0fec8fec1bb4cfef59e1f9ccedde4a5/models/spring-ai-openai/src/main/java/org/springframework/ai/openai/api/OpenAiApi.java#L671):
> .uri("/v1/chat/completions")

In this situation, third-party APIs that have a url pattern below cannot be used with Spring AI but can with OpenAI's python package.
> 3rd party's base url: https://foo.com/v1/bar
> 3rd party's path: /chat/completions 

Is there any plan to move version to base url or is  there any alternatives?

**Comment by javamagong** [comment]
It have been support config base-url 
![image](https://github.com/spring-projects/spring-ai/assets/1988270/365e6227-82b3-4bc4-aca5-f494a6f81eef)

**Reply by intotherealworld** [reply]
Thank you for your reply.
But, I know that you answered and It's not a solution I want.
As I mentioned above, if a API URL is like https://api.foo.com/v1/bar/chat/completions, there is no way to use it. However, the python package of OpenAPI is possible because it doesn't add /v1/ prefix to uri path.

**Comment by kiview** [comment, marked as answer]
You can use `spring.ai.openai.chat.completions-path`, see https://docs.spring.io/spring-ai/reference/api/chat/openai-chat.html#_chat_properties.

**Reply by intotherealworld** [reply]
Thank you for your reply.
This issue has already been resolved as of version 1.0.0. I was aware of this but unfortunately forgot to close the issue. I appreciate your reminder and your helpful response.
